test/src/e2e.dynval/dv.n-1.test.ts

```diff
@@ -0,0 +1,155 @@
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import * as chai from "chai";
+import { expect } from "chai";
+import * as chaiAsPromised from "chai-as-promised";
+import * as stake from "codechain-stakeholder-sdk";
+import { SDK } from "codechain-sdk";
+import "mocha";
+
+import { validators as originalValidators } from "../../tendermint.dynval/constants";
+import { faucetAddress, faucetSecret } from "../helper/constants";
+import { PromiseExpect } from "../helper/promise";
+import { withNodes } from "./setup";
+
+chai.use(chaiAsPromised);
+
+const alice = originalValidators[0];
+const otherDynValidators = originalValidators.slice(1, 8);
+const allDynValidators = [alice, ...otherDynValidators];
+
+describe("Dynamic Validator N -> N-1", function() {
+    this.timeout(0);
+
+    const promiseExpect = new PromiseExpect();
+    const termSeconds = 20;
+    const margin = 1.2;
+
+    async function aliceDropOutCheck(sdk: SDK) {
+        const blockNumber = await sdk.rpc.chain.getBestBlockNumber();
+        const termMedata = await stake.getTermMetadata(sdk, blockNumber);
+        const currentTermInitialBlockNumber =
+            termMedata!.lastTermFinishedBlockNumber + 1;
+        const validatorsAfter = (await stake.getPossibleAuthors(
+            sdk,
+            currentTermInitialBlockNumber
+        ))!.map(platformAddr => platformAddr.toString());
+
+        expect(termMedata!.currentTermId).to.be.equals(2);
+        expect(validatorsAfter.length).to.be.equals(otherDynValidators.length);
+        expect(validatorsAfter).not.to.includes(
+            alice.platformAddress.toString()
+        );
+        expect(
+            validatorsAfter!.map(platformAddr => platformAddr.toString())
+        ).contains.all.members(
+            otherDynValidators.map(validator => validator.platformAddress.value)
+        );
+    }
+
+    describe("A node is imprisoned to jail", async function() {
+        const nodes = withNodes(this, {
+            promiseExpect,
+            validators: allDynValidators.map((signer, index) => ({
+                signer,
+                delegation: 5_000,
+                deposit: 10_000_000 - index // tie-breaker
+            })),
+            onBeforeEnable: async allDynNodes => {
+                const aliceNode = allDynNodes[0];
+                // Kill the alice node first to make alice not to participate in the term 1.
+                await aliceNode.clean();
+            }
+        });
+
+        it("alice should be dropped out from validator list", async function() {
+            const checkingNode = nodes[1];
```

Comment:
Let's check Alice was in the validator list for the first time.

Reply:
I added the routine.